src/panels/ActionsPanel/ActionsPanel.tsx

```diff
@@ -125,100 +29,109 @@ export function ActionsPanel() {
+  function pathBreadbrumbs(): React.JSX.Element {
+    const paths = displayedNavigationPath.split('/');
     return (
-      <Button onClick={() => addNavPath(key)} key={key}>
-        <FolderIcon />
-        <p>{key}</p>
-      </Button>
+      <Breadcrumbs separatorMargin={0}>
+        {paths.map((path, i) => (
+          <Button
+            key={`${path}_${i}`}
+            p={2}
+            variant={'subtle'}
+            onClick={() => goToPath(path)}
+          >
+            {path}
+          </Button>
+        ))}
+      </Breadcrumbs>
     );
   }
 
-  function getFoldersContent(level: FolderContent) {
-    return Object.keys(level.folders)
-      .sort()
-      .map((key) => folderButton(key));
-  }
-
-  function getActionContent(level: FolderContent) {
-    if (level.actions.length === 0) {
-      return null;
-    }
-
-    return level.actions.map((action: Action) => (
-      <ActionsButton key={`${action.identifier}Action`} action={action} />
-    ));
-  }
-
-  function getBackButton() {
-    if (navigationPath !== '/') {
-      return (
-        <Button onClick={goBack} key={'backbtn'}>
-          Back
-        </Button>
-      );
-    }
-    return null;
-  }
-
-  if (actionLevel === undefined) {
+  if (!isInitialized || !actionLevel) {
     return <p>Loading...</p>;
   }
-  // TODO (ylvse 2024-10-10) Check what this check is meant to do?
-  // In the js it is actionlevel.length === 0. But how can it have a length?
-  const isEmpty = actionLevel === null;
-  const actions = isEmpty ? <div>No Actions</div> : getActionContent(actionLevel);
-  const folders = isEmpty ? <div>No Folders</div> : getFoldersContent(actionLevel);
-  const backButton = getBackButton();
+
   return (
     <ScrollArea h={'100%'}>
-      <Flex>
-        {backButton}
-        <p>{`${displayedNavigationPath}`}</p>
-      </Flex>
-      <FilterList>
-        <FilterList.InputField placeHolderSearchText={'Search for an action...'} />
-        <FilterList.Favorites>
-          <Grid>
-            {folders}
-            {actions}
-          </Grid>
-        </FilterList.Favorites>
-        <FilterList.Data<Action>
-          data={displayedActions}
-          renderElement={(action: Action) => (
-            <ActionsButton key={`${action.identifier}Filtered`} action={action} />
-          )}
-          matcherFunc={generateMatcherFunctionByKeys(['identifier', 'name', 'guiPath'])}
-        />
-      </FilterList>
+      <Container mt={'xs'}>
+        {!isTopLevel && (
+          <Group gap={'xs'} mb={'xs'}>
+            <ActionIcon onClick={goBack}>
+              <BackArrowIcon />
+              <VisuallyHidden>Back</VisuallyHidden>
+            </ActionIcon>
+            {pathBreadbrumbs()}
+          </Group>
+        )}
+        <FilterList>
+          <FilterList.InputField placeHolderSearchText={'Search for an action...'} />
+          <FilterList.Favorites>
+            <DynamicGrid gutter={'xs'} minChildSize={150} className={'actionsPanelGrid'}>
```

Comment:
This component should take a prop that is the width so it's not only using the window width:)